crates/router/src/core/routing/helpers.rs

```diff
@@ -740,18 +740,36 @@ pub async fn push_metrics_with_update_window_for_success_based_routing(
             .label
             .to_string();
 
-        let (first_success_based_connector, _) = first_success_based_connector_label
+        let (first_merchant_success_based_connector, _) = first_merchant_success_based_connector_label
             .split_once(':')
             .ok_or(errors::ApiErrorResponse::InternalServerError)
             .attach_printable(format!(
                 "unable to split connector_name and mca_id from the first connector {:?} obtained from dynamic routing service",
-                first_success_based_connector_label
+                first_merchant_success_based_connector_label
+            ))?;
+
+        let first_global_success_based_connector_label = &success_based_connectors
+            .global_scores_with_labels
+            .first()
+            .ok_or(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable(
+                "unable to fetch the first connector from list of connectors obtained from dynamic routing service",
+            )?
+            .label
+            .to_string();
+
+        let (first_global_success_based_connector, _) = first_global_success_based_connector_label
+            .split_once(':')
+            .ok_or(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable(format!(
+                "unable to split connector_name and mca_id from the first connector {:?} obtained from dynamic routing service",
+                first_global_success_based_connector_label
             ))?;
 
         let outcome = get_success_based_metrics_outcome_for_payment(
             payment_status_attribute,
             payment_connector.to_string(),
-            first_success_based_connector.to_string(),
+            first_merchant_success_based_connector.to_string(),
         );
 
         let dynamic_routing_stats = DynamicRoutingStatsNew {
```

Comment:
should we create a new column here which stores the `globally_top_performing_connector` at the time this particular payment was made?
cc: @prajjwalkumar17